### DISCUSSION TITLE
What's the Best Way to Update Valhalla & Tiles

### DISCUSSION THREAD

**Discussion by mertbakir** [valhalla/valhalla#4393] [Q&A] [answered]
Hi, I wonder what's the recommended way to update the tiles? Rendering the tiles from the osm.pbf files takes so time. So, I was thinking

- Spinning up a new valhalla instance with the up-to-date osm.pbf files
- Wait for it the process the files
- Point the requests to the new instance
- Delete the old instance

This requires some effort though. So, I wonder if there is a simpler way. 

Can Valhalla process new tiles while still being available for the incoming requests and respond using the old/existing tiles?

**Comment by kevinkreiser** [comment]
> Can Valhalla process new tiles while still being available for the incoming requests and respond using the old/existing tiles?

though we have made some progress toward supporting that, at the moment it cannot load new tiles without a restart. it is possible for us to write some code to periodically re-memory map the tar file that is on the disk assuming that someone overwrote it with a new one (ie. `mv net_tiles.tar old_tiles.tar`). we just need a bit of code that periodically re validates whats on disk is what we have memory mapped and if not we re-do the mapping. its not a trivial effort but its not too difficult either.

**Reply by mertbakir** [reply]
Would you say the way I'm thinking to update (spinning up a new instance) makes sense at the moment?

**Reply by kevinkreiser** [reply, marked as answer]
yes, this strategy is called blue-green deployment and it is the most commonly used way to update a running valhalla service

**Reply by mertbakir** [reply]
thanks
